src/receivers/HTTPReceiver.ts

```diff
@@ -506,91 +427,49 @@ export default class HTTPReceiver implements Receiver {
       // req object, so that its as reusable as possible. Later, we should consider adding an option for assigning the
       // parsed body to `req.body`, as this convention has been established by the popular `body-parser` package.
       try {
-        body = parseBody(bufferedReq);
+        body = httpFunc.parseHTTPRequestBody(bufferedReq);
       } catch (err) {
         const e = err as any;
         this.logger.warn(`Malformed request body: ${e.message}`);
-        res.writeHead(400);
-        res.end();
+        httpFunc.buildNoBodyResponse(res, 400);
         return;
       }
 
       // Handle SSL checks
       if (body.ssl_check) {
-        res.writeHead(200);
-        res.end();
+        httpFunc.buildNoBodyResponse(res, 200);
         return;
       }
 
       // Handle URL verification
       if (body.type === 'url_verification') {
-        res.writeHead(200, { 'content-type': 'application/json' });
-        res.end(JSON.stringify({ challenge: body.challenge }));
+        httpFunc.buildUrlVerificationResponse(res, body);
         return;
       }
 
-      // Setup ack timeout warning
-      let isAcknowledged = false;
-      setTimeout(() => {
-        if (!isAcknowledged) {
-          this.unhandledRequestHandler({
-            logger: this.logger,
-            request: req,
-            response: res,
-          });
-        }
-      }, this.unhandledRequestTimeoutMillis);
-
+      const ack = new HTTPResponseAck({
```

Comment:
Some people may not like the idea to have a class object with state in general but we have many state / local variables (isAcknowledged, storedResponse, bufferedReq, res) here. Managing them in a single place should not be a bad idea.